src/NHibernate/Type/AbstractStringType.cs

```diff
@@ -58,13 +58,16 @@ public override void Set(DbCommand cmd, object value, int index, ISessionImpleme
 		{
 			var parameter = cmd.Parameters[index];
 
+			if (value is Enum)
+				value = value.ToString();
+
 			//Allow the driver to adjust the parameter for the value
 			session.Factory.ConnectionProvider.Driver.AdjustParameterForValue(parameter, SqlType, value);
 
 			// set the parameter value before the size check, since ODBC changes the size automatically
 			parameter.Value = value;
 
-			if (parameter.Size > 0 && ((string)value).Length > parameter.Size)
+			if (parameter.Size > 0 && value != null && ((string) value).Length > parameter.Size)
```

Comment:
Undue change. `Set` is not supposed to be called with a `null` value. It is called by `NullSafeSet` only if the value is not null.

Reply:
```suggestion
			if (parameter.Size > 0 && ((string)value).Length > parameter.Size)
```

Reply:
@fredericDelaporte reverted.